pyuvdata/uvdata/uvdata.py

```diff
@@ -7145,8 +7147,25 @@ def read(
                         strict_uvw_antpos_check=strict_uvw_antpos_check,
                     )
                     unread = False
-                except KeyError:
-                    warnings.warn("Failed to read {f}".format(f=filename[file_num]))
+                except KeyError as err:
+                    file_warnings = (
+                        file_warnings + f"Failed to read {f} due to KeyError: {err}\n"
+                    )
+                    file_num += 1
+                    if skip_bad_files is False:
+                        raise
+                except ValueError as err:
+                    file_warnings = (
+                        file_warnings + f"Failed to read {f} due to ValueError: {err}\n"
+                    )
+                    file_num += 1
+                    if skip_bad_files is False:
+                        raise
+                # pragma: nocover
```

Comment:
This needs to be on the next line just after the colon to work properly (it tells the coverage calculator to ignore that line, and since it's an except block, to also ignore the lines inside the block).